s3/s3.go

```diff
@@ -0,0 +1,172 @@
+package s3
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	bindings "github.com/andream16/aws-sdk-go-bindings"
+	"github.com/andream16/aws-sdk-go-bindings/internal/format"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/pkg/errors"
+)
+
+// S3er describes s3 API.
+type S3er interface {
+	CreateBucket(string) error
+	GetObject(bucket, path string) ([]byte, error)
+	PutObject(bucket, objectName, objectPath string) error
+}
+
+// S3 is the alias for s3.
+type S3 struct {
+	s3 s3iface.S3API
+}
+
+type file struct {
+	body          []byte
+	contentType   string
+	contentLength int64
+}
+
+// New returns a new S3.
+func New(region string, options ...bindings.Option) (*S3, error) {
+
+	if region == "" {
+		return nil, errors.New("required parameter region cannot be empty")
```

Comment:
Minor: there's a mix-match of dynamic errors and defined errors, e.g. this one can't be interrogated by a caller of the package, but in `CreateBucket` I could test against `bindings.ErrInvalidParameter` - in fact I'd say this error qualifies to use the `bindings.ErrInvalidParameter` error with wrapped context?